system/datatypes/peer_reviewed_article.php

```diff
@@ -1233,14 +1363,53 @@ private function sendNewCommentMail(HyphaDomElement $comment) {
 			'name' => $name,
 			'title' => $title
 		]);
-		$message = __('art-comment-body', [
+		$messageArgs = [
 			'name' => htmlspecialchars($name),
 			'link' => htmlspecialchars($linkToPage),
 			'title' => htmlspecialchars($title),
 			'comment' => nl2br(htmlspecialchars($commentBody)),
+		];
+		$message = __('art-comment-body', $messageArgs);
+		$this->sendMail(getUserEmailList(), $subject, $message);
+
+		$subscriberSubject = __('art-comment-subscriber-subject', [
+			'name' => $name,
+			'title' => $title
 		]);
+		$subscriberMessage = __('art-comment-subscriber-body', $messageArgs);
+
+		// also send newly create comment to all who subscribed to this discussion, with unsubscribe link
+		foreach ($this->getSubscribers($comment) as $subscriber) {
+			$email = $subscriber->getAttribute(self::FIELD_NAME_DISCUSSION_SUBSCRIBER_EMAIL);
+			$code = $subscriber->getAttribute(self::FIELD_NAME_DISCUSSION_SUBSCRIBER_UNSUBSCRIBE_CODE);
+			// include email for server record purposes
+			$path = hypha_substitute(self::PATH_UNSUBSCRIBE_CODE, ['address' => htmlspecialchars($email), 'code' => htmlspecialchars($code)]);
```

Comment:
I think this should not do `htmlspecialchars`, if anything it should do `rawurlencode` (since you're building a URL here. The `htmlspecialchars` should be done below, on the full `$linkToUnsubscribe` when it is inserted into the `href` attribute.

Reply:
fixed